texar/torch/modules/tokenizers/pretrained_bert_tokenizer.py

```diff
@@ -0,0 +1,199 @@
+                                      self._VOCAB_FILE_NAMES['vocab_file'])
+            assert pretrained_model_name is not None
+            if self._MAX_INPUT_SIZE.get(pretrained_model_name):
+                self.max_len = self._MAX_INPUT_SIZE[pretrained_model_name]
+        else:
+            vocab_file = self.hparams['vocab_file']
+            if self.hparams.get('max_len'):
+                self.max_len = self.hparams['max_len']
+
+        if not os.path.isfile(vocab_file):
+            raise ValueError("Can't find a vocabulary file at path "
+                             "'{}".format(vocab_file))
+        self.vocab = load_vocab(vocab_file)
+        self.ids_to_tokens = dict((ids, tok) for tok, ids in self.vocab.items())
+
+        self.do_basic_tokenize = self.hparams['do_basic_tokenize']
+        if self.do_basic_tokenize:
+            self.basic_tokenizer = BasicTokenizer(
+                do_lower_case=self.hparams["do_lower_case"],
+                never_split=self.hparams["never_split"],
+                tokenize_chinese_chars=self.hparams["tokenize_chinese_chars"])
+        self.wordpiece_tokenizer = WordpieceTokenizer(vocab=self.vocab,
+                                                      unk_token=self.unk_token)
+
+    def forward(self, inputs, task, **kwargs):
+        if task == 'text-to-token':
+            return self.tokenize(inputs)
+        elif task == 'token-to-text':
+            return self.convert_tokens_to_string(inputs)
+        elif task == 'text-to-id':
+            return self.encode(inputs)
+        elif task == 'id-to-text':
+            skip_special_tokens = kwargs.get('skip_special_tokens', False)
+            clean_up_tokenization_spaces = kwargs.get(
+                'clean_up_tokenization_spaces', True)
+            return self.decode(
+                inputs, skip_special_tokens=skip_special_tokens,
+                clean_up_tokenization_spaces=clean_up_tokenization_spaces)
+        elif task == 'token-to-id':
+            return self.convert_tokens_to_ids(inputs)
+        elif task == 'id-to-token':
+            skip_special_tokens = kwargs.get('skip_special_tokens', False)
+            return self.convert_ids_to_tokens(
+                inputs, skip_special_tokens=skip_special_tokens)
+        else:
+            raise ValueError("Unrecognized task.")
+
+    def _tokenize(self, text: str) -> List[str]:  # type: ignore
+        split_tokens = []
+        if self.do_basic_tokenize:
+            for token in self.basic_tokenizer.tokenize(
+                    text, never_split=self.all_special_tokens):
+                assert token is not None
+                for sub_token in self.wordpiece_tokenizer.tokenize(token):
+                    split_tokens.append(sub_token)
+        else:
+            split_tokens = self.wordpiece_tokenizer.tokenize(text)
+        return split_tokens
+
+    def save_vocabulary(self, save_directory: str) -> Tuple[str]:
```

Comment:
name it as `save_vocab` since other places also refer to vocabulary as "vocab", e.g., `vocab_size`